pandas/tests/frame/methods/test_asfreq.py

```diff
@@ -283,3 +281,19 @@ def test_asfreq_unsupported_freq(self, freq, error_msg):
 
         with pytest.raises(ValueError, match=error_msg):
             df.asfreq(freq=freq)
+
+    @pytest.mark.parametrize(
+        "freq, freq_depr",
+        [
+            ("1YE", "1A"),
+            ("2YE-MAR", "2A-MAR"),
+        ],
+    )
+    def test_asfreq_frequency_A_raisees(self, freq, freq_depr):
+        msg = f"Invalid frequency: {freq_depr[1:]}"
+
+        index = date_range("1/1/2000", periods=4, freq=f"{freq[1:]}")
+        df = DataFrame({"s": Series([0.0, 1.0, 2.0, 3.0], index=index)})
+
+        with pytest.raises(ValueError, match=msg):
+            df.asfreq(freq=freq_depr)
```

Comment:
it's not clear to me what you're doing with `freq` in this test - I think `index` can just be instatiated with `'YE'` in both cases?

Reply:
yes, you are right, it's sufficient to check only `freq="A"`. I simplified the test and added a note to `v3.0.0`.
Could you please take a look at my changes?